source/Octo.Tests/Commands/CommandDesignFixture.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using FluentAssertions;
+using NUnit.Framework;
+using Octopus.Cli.Commands;
+using Octopus.Cli.Infrastructure;
+
+namespace Octo.Tests.Commands
+{
+    [TestFixture]
+    public class CommandDesignFixture
+    {
+        [Test]
+        public void ShouldAttachWithCorrectAttribute()
+        {
+            var commandClassNamesWithoutCorrectAttribute = Assembly.GetAssembly(typeof(ICommand)).GetTypes()
+                .Where(t => typeof(ICommand).IsAssignableFrom(t) &&
+                            t.GetCustomAttribute<CommandAttribute>() == null &&
+                            !t.IsAbstract &&
+                            !t.IsInterface)
+                .Select(t => t.Name)
+                .ToList()
+                .AsReadOnly();
+
+            commandClassNamesWithoutCorrectAttribute.Should().BeEmpty($"Each command which implements '{nameof(ICommand)}' interface must be attached with '{nameof(CommandAttribute)}'. " +
+                                                                      $"The following command classes: ({string.Join(", ", commandClassNamesWithoutCorrectAttribute)}) are not good enough for you?");
+        }
+
+        [Test]
+        public void SubClassesOfApiCommand_ShouldEitherOverrideExecuteMethodOrImplementCorrectInterface()
+        {
+            var commandTypes = Assembly.GetAssembly(typeof(ICommand)).GetTypes()
```

Comment:
This block is largely repeated. Would it be better to make the types that implement `ICommand` a `TestCaseSource` and have have a test per type? I think it would be much clearer in the output which Commands failed to pass. We have examples of this in `Octopus.Client`.

If not, I'd move this out into a `GetTypesThatImplementICommand` method.